warehouse/admin/views/banners.py

```diff
@@ -0,0 +1,27 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyramid.view import view_config
+
+from warehouse.banners.models import Banner
+
+
+@view_config(
+    route_name="admin.banner.list",
+    renderer="admin/banners/list.html",
+    permission="moderator",
+    request_method="GET",
+    uses_session=True,
+)
+def banner_list(request):
+    banners = request.db.query(Banner).order_by(Banner.begin.desc()).all()
```

Comment:
I'd recommend making deletion of banners accessible via the admin UI (if it is not already). I'd actually assume that most banners will be deleted rather than left to sit rotting.

Reply:
Yes, we already have delete support and I think I'm with @ewdurbin in this one. 

**Or**

If we think it's worthwhile, we ca set up a monthly cron job to clean up older and expired banners (with end date from 3 months ago)

I tend to like most the second alternative since it address both problems: prevent undesired slow DB performance without extra human review work.